pkg/api/types.go

```diff
@@ -735,6 +740,10 @@ type MultiStageTestConfiguration struct {
 	// all previous `pre` and `test` steps were successful. The given step must explicitly
 	// ask for being skipped by setting the OptionalOnSuccess flag to true.
 	AllowSkipOnSuccess *bool `json:"allow_skip_on_success,omitempty"`
+	// AllowBestEffortPostSteps defines if any `post` steps can be ignored when
+	// they fail. The given step must explicitly ask for being ignored by setting
+	// the OptionalOnSuccess flag to true.
+	AllowBestEffortPostSteps *bool `json:"allow_best_effort_post_steps,omitempty"`
```

Comment:
Is this flag necessary given that the default for `BestEffort` is `false`?

Reply:
Without it, a step becomes best-effort on all jobs when it's configuration is changed. Is that what we want?

Reply:
Ah, make sense, that's what was trying to understand.